### PR TITLE
Adrv9002 variants support

### PR DESCRIPTION
**Pull request by tagoylo**
This added support to handle adrv9002 variants. Board names with -v<variant> at the end of their board name are parsed, getting the <variant> to be passed as a marker to pyadi-iio test. 

This also adds a feature that ensures a resource having different variants will only be accessed and used one variant at a time, this is by using the lock feature.

This feature is dependent on nebula config and nebula.

### CODE REVIEW

File: src/sdg/Gauntlet.groovy [hunk 1]
```diff
@@ -1113,6 +1160,31 @@ private def get_gitsha(String board){
     }
 }
 
+private def check_for_marker(String board){
+    if (board.contains("-v")){
+        board_name = board.split("-v")[0]
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:1165]: might need to put 'def' on these variables, so scope is just within the method.

File: src/sdg/Gauntlet.groovy [hunk 2]
```diff
@@ -1113,6 +1160,31 @@ private def get_gitsha(String board){
...
+        marker = board.split("-v")[1]
+        return [board_name:board_name, marker:marker]
+    }
+    else {
+        board_name = board
+        return board_name
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:1171]: consider returning the same map structure as the first condition, 
ex. return [board_name:board_name, marker:none]

the following calls might issue error if check is not a map with attribues board_name and marker
board_name = check.board_name.replaceAll('-', '_')
marker = check.marker

File: src/sdg/Gauntlet.groovy [hunk 3]
```diff
@@ -431,12 +433,15 @@ def stage_library(String stage_name) {
...
                             if (gauntEnv.iio_uri_source == "ip")
                                 uri = "ip:" + ip;
                             else
                                 uri = "serial:" + serial + "," + gauntEnv.iio_uri_baudrate.toString()
-                            cmd = "python3 -m pytest --html=testhtml/report.html --junitxml=testxml/" + board + "_reports.xml --adi-hw-map -v -k 'not stress' -s --uri='"+uri+"' -m " + board + " --capture=tee-sys"
+                            check = check_for_marker(board)
+                            board = board.replaceAll('-', '_')
+                            board_name = check.board_name.replaceAll('-', '_')
+                            marker = check.marker
+                            cmd = "python3 -m pytest --html=testhtml/report.html --junitxml=testxml/" + board + "_reports.xml --adi-hw-map -v -k 'not stress' -s --uri='ip:"+ip+"' -m " + board_name + " --capture=tee-sys" + " --" + marker
```
Comment (kimpaller) [inline src/sdg/Gauntlet.groovy:445]: might necessary to remove '+ --' when marker is empty

File: src/sdg/Gauntlet.groovy (hunk 1 above)
Comment (tagoylo) [reply src/sdg/Gauntlet.groovy:1167]: fixed with commit 3a31db4.

File: src/sdg/Gauntlet.groovy (hunk 2 above)
Comment (tagoylo) [reply src/sdg/Gauntlet.groovy]: fixed with commit 3a31db4.

File: src/sdg/Gauntlet.groovy (hunk 3 above)
Comment (tagoylo) [reply src/sdg/Gauntlet.groovy]: fixed with commit 3a31db4. Added to marker variable in check_for_marker method.

**Pull request merged**